src/coreclr/jit/fgopt.cpp

```diff
@@ -419,6 +420,9 @@ bool Compiler::fgRemoveUnreachableBlocks()
             /* Unmark the block as removed, */
             /* clear BBF_INTERNAL as well and set BBJ_IMPORTED */
 
+            // The successors may be unreachable after this change.
+            changed |= block->NumSucc() > 0;
```

Comment:
can it be just `hasUnreachableBlocks |= block->NumSucc() > 0;` without that additional argument? 
the `if (hasUnreachableBlocks)` block won't do anything for `BBF_DONT_REMOVE` blocks anyway

Reply:
Seems clearer to me to keep the two separate.

Reply:
I went back and forth on this, but decided to go with the extra variable since the loop below doesn't need to run unless there are actually unreachable blocks.